### PR TITLE
add js-waku encryption guide

### PR DESCRIPTION
**Pull request by LordGhostX**
Tackles some of the tasks in https://github.com/waku-org/docs.waku.org/issues/125

preview: https://docs-waku-org-git-js-encryption-acidinfo.vercel.app/guides/js-waku/message-encryption

pending:

- [ ] merging https://github.com/waku-org/js-waku-examples/pull/286
- [ ] release of @waku/sdk with `verifySignature()`

### CODE REVIEW

File: docs/guides/js-waku/message-encryption.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+// Generate a public key from the private key for verifying signatures
+const sigPubKey = getPublicKey(sigPrivKey);
+
+// Create an encoder that signs messages
+const encoder = createEncoder({
+	contentTopic: contentTopic,
+	symKey: symKey,
+	sigPrivKey: sigPrivKey,
+});
```
Comment (fryorcraken) [inline docs/guides/js-waku/message-encryption.md:190]: I think for this example to talk to the reader, we need to introduce Alice/Bob or local/remote.

File: docs/guides/js-waku/message-encryption.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	// Compare the public key of the message signature with the sender's own
+	if (JSON.stringify(signaturePublicKey) === JSON.stringify(sigPubKey)) {
+		console.log("This message was correctly signed");
+	} else {
+		console.log("This message has an incorrect signature");
+	}
+};
+```
+
+## Restoring encryption keys
```
Comment (fryorcraken) [inline docs/guides/js-waku/message-encryption.md:207]: I am not sure this is necessary//useful.

However, do feel free to use https://github.com/waku-org/js-waku-examples/tree/master/examples/eth-pm/src/key_pair_handling as an example on how to save/restore key information from local storage using proper security (Subtle Crypto)

File: docs/guides/js-waku/message-encryption.md (hunk 1 above)
Comment (LordGhostX) [reply docs/guides/js-waku/message-encryption.md]: Screenshot 2023-12-08 at 13 43 54

went with the Alice/Bob reference

File: docs/guides/js-waku/message-encryption.md (hunk 2 above)
Comment (LordGhostX) [reply docs/guides/js-waku/message-encryption.md:212]: I've added the `key_pair_handling` example, but I still think the `bytesToHex` and `hexToBytes` examples are helpful for some key exporting use cases. Let me know what you think about the update

File: docs/guides/js-waku/message-encryption.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		console.log("This message was NOT signed by Alice");
+	}
+};
+
+await subscription.subscribe([encoder], callback);
+```
+
+## Storing encryption keys
+
+We used randomly generated keys for encryption and message signing in the provided examples, but real-world applications require consistent keys among clients. Have a look at the [Key Pair Handling](https://github.com/waku-org/js-waku-examples/tree/master/examples/eth-pm/src/key_pair_handling) example, which demonstrates the secure storage and retrieval of key information from local storage using [Subtle Crypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto).
```
Comment (vpavlin) [inline docs/guides/js-waku/message-encryption.md:212]:
```suggestion
We used randomly generated keys for encryption and message signing in the provided examples, but real-world applications require consistent keys among client restarts. Have a look at the [Key Pair Handling](https://github.com/waku-org/js-waku-examples/tree/master/examples/eth-pm/src/key_pair_handling) example, which demonstrates the secure storage and retrieval of key information from local storage using [Subtle Crypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto).
```

**Review by vpavlin — APPROVED** [review]
Left one comment, otherwise looks great to me!

File: docs/guides/js-waku/message-encryption.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+You can also use the [@waku/utils](https://www.npmjs.com/package/@waku/utils) package to store keys in hexadecimal format:
```
[the remark this answers is not in the document]
Comment (LordGhostX) [reply docs/guides/js-waku/message-encryption.md:214]: I think that's a great suggestion, I've implemented it 👍

**Pull request merged**